lib/elven_gard/packet_encoders/binary_encoder.ex

```diff
@@ -0,0 +1,97 @@
+defmodule ElvenGard.PacketEncoders.BinaryEncoder do
+  @moduledoc false
+
+  alias ElvenGard.Structures.{Client, PacketDefinition}
+
+  @aliases [
+    byte: ElvenGard.Types.Binary.ByteType,
```

Comment:
Why use `Types` namespace and re-add a `Type` suffix on the sub-module?

Reply:
To avoid conflicts (`String` for exemple)

Reply:
So you should have namespaced it like `ElvenGard.Binary.ByteType`.

Reply:
It sounds weird to have an `ElvenGard.Binary` module and an `ElvenGard.Textual module without them being in a parent namespace themselves. They seem to come out of nowhere and we don't know what we're going to find in them

Reply:
Or the most common way to handle it is to consider these 2 modules as a kind of adapter like:
```
ElvenGard.Protocol (@behaviour)
ElvenGard.Protocol.Binary
ElvenGard.Protocol.Text
```
Such as: `Ecto.Adapter`

Reply:
Ok, I like that